android/modules/ui/src/java/ti/modules/titanium/ui/WebViewProxy.java

```diff
@@ -110,9 +119,94 @@ public Object evalJS(String code)
 			Log.w(TAG, "WebView not available, returning null for evalJS result.");
 			return null;
 		}
+		if (callback != null) {
+			// When on Android 19+ we can use the builtin evalAsync method!
+			if (Build.VERSION.SDK_INT >= 19) {
+				final KrollFunction theCallback = callback;
+				view.getWebView().evaluateJavascript(code, new ValueCallback<String>() {
```

Comment:
Google documents that `evaluateJavascript()` must be called on the main UI thread here...
[https://developer.android.com/reference/android/webkit/WebView.html#evaluateJavascript](https://developer.android.com/reference/android/webkit/WebView.html#evaluateJavascript(java.lang.String,%20android.webkit.ValueCallback<java.lang.String>))

I looked at Google's source code on github. It'll throw an exception if invoked on any other thread.
https://github.com/aosp-mirror/platform_frameworks_base/blob/master/core/java/android/webkit/WebView.java#L1053